app/forms/instance-create.tsx

```diff
@@ -278,10 +284,18 @@ export function CreateInstanceForm() {
         required={false}
         control={control}
         disabled={isSubmitting}
+        validate={(name) => {
+          // don't allow the user to use an existing disk name for the boot disk's name
+          if ([...namesOfDisksToBeCreated, ...existingDiskNames].includes(name)) {
```

Comment:
Very small thing, but I think it would be easier to follow if this was done next to the other disk name list logic. You can even wrap it in a function to avoid mapping over a potentially large list of disks on every render.

```ts
const otherDisks = useWatch({ control, name: 'otherDisks' })

function unavailableDiskNames() {
  const existingDiskNames = allDisks.map((disk) => disk.name)
  const namesOfDisksToBeCreated = otherDisks
    .filter((disk) => disk.type === 'create')
    .map((disk) => disk.name)
  return [...namesOfDisksToBeCreated, ...existingDiskNames]
}
```

Reply:
I just confirmed that typing in the disk name field of disk create does not render the overall form, so it is better to leave it as-is.